test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj

```diff
@@ -63,7 +63,9 @@
   (testing "valid-code-id? accepts valid code-ids"
     (is (vc-core/valid-code-id? nil))
     (is (vc-core/valid-code-id? "dcf16ec"))
-    (is (vc-core/valid-code-id? "master-plan:stage_1:destroy-all-humans")))
+    (is (vc-core/valid-code-id? "whatMakesJavaBad;isPartlySemiColons;clojureIsRealNice"))
+    (is (vc-core/valid-code-id? "master-plan:stage_1:destroy-all-humans"))
+    (is (vc-core/valid-code-id? "combining:lots;of_valid-characters")))
```

Comment:
@jpinsonault wanna add one more example to this - a URN example like Josh posted in his comment on the previous PR?